subsys/nrf_security/src/utils/nrf_security_events.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
```

Comment:
A comment explaining why this exists would be helpful for a future reader I think.

Something like

This event interface is inspured by Zephyr's event interface. There exists two implementations here. One is backed by Zephyr EVENTS, and other assumes there are no threads and uses cortex-m33 __WFE instructions.

It exists to support three scenarios. TF-M, which doesnt use multithreading. Zephyr with multithreading enabled, and Zephyr with multithreading disabled (bootloaders typically).

Reply:
I used your comment thanks, I was planning to add something but glad that you wrote it for me :)